tcms/rpc/tests/test_testexecution.py

```diff
@@ -274,6 +276,78 @@ def test_get_links(self):
         self.assertEqual(logs[0]['url'], 'http://kiwitcms.org')
 
 
+class TestExecutionHistory(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.execution = TestExecutionFactory()
+
+    def test_history_for_non_existing_execution(self):
+        with self.assertRaisesRegex(XmlRPCFault, "TestExecution matching query does not exist."):
+            self.rpc_client.TestExecution.history(-5)
+
+    def test_history_new_execution(self):
+        execution = TestExecutionFactory()
+
+        history = self.rpc_client.TestExecution.history(execution.pk)
+
+        self.assertEqual(1, len(history))
+
+    def test_history(self):
+        """
+        Test that for an execution that has been updated 3 times,
+        there are 4 history entries (first one is the creation of the object).
+
+        Note: the `time.sleep` call after every update is necessary,
+        because otherwise the changes happen too fast,
+        and the measurable delta is 1 second.
+        """
+        time.sleep(1)
+
+        build = BuildFactory()
+        self.execution.build = build
+        self.execution.save()
+        time.sleep(1)
+
+        user = UserFactory()
+        self.execution.assignee = user
+        self.execution.save()
+        time.sleep(1)
+
+        self.execution.tested_by = user
+        self.execution.save()
+        time.sleep(1)
+
+        history = self.rpc_client.TestExecution.history(self.execution.pk)
+        self.assertEqual(4, len(history))
+
+        # assert entries are in the right order
+        previous = None
```

Comment:
initialize with timezone.now() and then assign the timestamp to this temporary variable. No need for the if condition and the assertion should work.  Also use `assertLess`

Reply:
fixed